anydex/wallet/ethereum/eth_wallet.py

```diff
@@ -0,0 +1,238 @@
+import os
+import time
+
+from ipv8.util import fail, succeed
+from sqlalchemy import func, or_
+from web3 import Web3
+
+from anydex.wallet.cryptocurrency import Cryptocurrency
+from anydex.wallet.ethereum.eth_db import initialize_db, Key, Transaction
+from anydex.wallet.ethereum.eth_provider import AutoEthereumProvider, AutoTestnetEthereumProvider
+from anydex.wallet.wallet import Wallet, InsufficientFunds
+
+
+class EthereumWallet(Wallet):
+    """
+    This class is responsible for handling your Ethereum wallet.
+    """
+    TESTNET = False
```

Comment:
I'm not sure.

Reply:
Bitcoinlib is an odd example in all of this due to its structure. We can try implementing IOTA in the inheritance manner and see how it compares, but this solution does seem clean, because it could have also all been just one class with 3 extra if statements and additional __init__ parameter, and instead it has a small override.